### PR TITLE
Casting section added to code style guide. Issue: #693.

### PR DESCRIPTION
**Pull request by pavlicekl**

### CODE REVIEW

**Comment by fabianthoma** [comment]
ok to test

**Comment by fabianthoma** [comment]
ok to test

**Comment by lewisgoddard** [comment]
If @danrabbit is happy with the placement, then I'm happy with the code.

File: docs/code/reference.md [hunk 1]
```diff
@@ -140,6 +140,17 @@ than good.
 Also worth referring that there should be no Hungarian notation, and no mix of
 any kinds of notations.
 
+## Casting {#casting}
+
+Prevent using `as` keyword when casting as it might give `null` as result, which could be easily forgotten to check.
```
Comment (lewisgoddard) [inline docs/code/reference.md]: From an language point of view, it might be better off said "Avoid using..." instead.

Comment (pavlicekl) [inline docs/code/reference.md]: I agree, "Avoid using ..." sounds better.

**Pull request merged**